metaworld/envs/mujoco/sawyer_xyz/sawyer_peg_insertion_side_v2.py

```diff
@@ -0,0 +1,241 @@
+import numpy as np
+from gym.spaces import Box
+
+from metaworld.envs.env_util import get_asset_full_path
+from metaworld.envs.mujoco.sawyer_xyz.base import SawyerXYZEnv
+
+
+class SawyerPegInsertionSideEnvV2(SawyerXYZEnv):
+    """
+    Motivation for V2:
+        V1 was difficult to solve because the observation didn't say where
+        to insert the peg (the hole's location). Furthermore, the hole object
+        could be initialized in such a way that it severely restrained the
+        sawyer's movement.
+    Changelog from V1 to V2:
+        - (6/16/20) Added a 1 element vector to the observation. This vector
+            points from the end effector to the hole in the Y direction.
+            i.e. (self._state_goal - pos_hand)[1]
+        - (6/16/20) Used existing goal_low and goal_high values to constrain
+            the hole's position, as opposed to hand_low and hand_high
+    """
+    def __init__(self, random_init=False):
+        liftThresh = 0.11
+        hand_init_pos = (0, 0.6, 0.2)
+
+        goal_low = (-0.35, 0.5, 0.05)
+        goal_high = (-0.25, 0.8, 0.05)
+        hand_low = (-0.5, 0.40, 0.05)
+        hand_high = (0.5, 1, 0.5)
+        obj_low = (-0.1, 0.5, 0.02)
+        obj_high = (0.1, 0.7, 0.02)
+
+
+        super().__init__(
+            self.model_name,
+            hand_low=hand_low,
+            hand_high=hand_high,
+        )
+
+        self.init_config = {
+            'obj_init_pos': np.array([0, 0.6, 0.02]),
+            'hand_init_pos': np.array([0, .6, .2]),
+        }
+
+        self.goal = np.array([-0.3, 0.6, 0.05])
+
+        self.obj_init_pos = self.init_config['obj_init_pos']
+        self.hand_init_pos = self.init_config['hand_init_pos']
+
+        self.random_init = random_init
+        self.liftThresh = liftThresh
+        self.max_path_length = 150
+
+        self.hand_init_pos = np.array(hand_init_pos)
+
+        self.action_space = Box(
+            np.array([-1, -1, -1, -1]),
+            np.array([1, 1, 1, 1]),
+        )
+
+        self.obj_and_goal_space = Box(
+            np.hstack((obj_low, goal_low)),
+            np.hstack((obj_high, goal_high)),
+        )
+        self.goal_space = Box(np.array(goal_low), np.array(goal_high))
+
+        self.observation_space = Box(
+            np.hstack((self.hand_low, obj_low, np.array(goal_low) - self.hand_high)),
+            np.hstack((self.hand_high, obj_high, self.hand_high - np.array(goal_low))),
+        )
+
+        self.reset()
+
+    @property
+    def model_name(self):
+        return get_asset_full_path('sawyer_xyz/sawyer_peg_insertion_side.xml')
```

Comment:
in some PRS I notice that we create new XMLs for our environments. I'm not sure if you removed this from your prs, but if you create a new xml, remember to change this line of code here to reflect that. I just remembered that I had not done this.

Reply:
Yeah we've been creating new .xml's when splitting reach-pick-place into 3 envs, but otherwise we've stuck with the original xml